### DISCUSSION TITLE
Authorisation BluetoothLE

### DISCUSSION THREAD

**Discussion by skramovskiy** [airsdk/Adobe-Runtime-Support#2071] [General]
Help, I just can’t figure out the library and Bluetooth LE.ane on android
when starting the function BluetoothLE.service.requestAuthorisation() - returns true;
when starting the BluetoothLE.service.authorizationStatus() function - returns should_explain;

registered permissions in the descriptor.
But no dialog box comes up (as the [manua](https://docs.airnativeextensions.com/docs/bluetoothle/request-authorisation)l says)?
how to get authorization?

**Comment by skramovskiy** [comment]
through the android settings, changed the permission settings,
was able to achieve the NOT DETERMINED status
but I could not get authorization

**Comment by skramovskiy** [comment]
вопрос закрыт,  манифест из руководства надо редактировать....

**Comment by marchbold** [comment]
Hi, Make sure you have correctly added the usage strings to your info additions, if you don't add these you will encounter issues with the dialogs.

https://docs.airnativeextensions.com/docs/bluetoothle/add-the-extension

**Reply by marchbold** [reply]
Also you can get direct support from distriqt if you post on the bluetoothle extension forum: https://github.com/distriqt/ANE-BluetoothLE/issues

**Comment by pinter75** [comment]
Using 2017.07.14 on Android 8, I can never get the auth popup? Tried using the latest 4.2 core, and nothing. Code is tracing out that it is being asked for, but nothing pops up?

**Reply by marchbold** [reply]
Are you using the extension from 2017? I'd doubt that would still work with the recent Android and AIR changes. I suggest you update to the latest release.

**Reply by pinter75** [reply]
It's odd, a version works on my Oppo5 (pre-compiled apk, built with an older air sdk), which is running a much higher Android version. A newly compiled version of the same code base, plugins etc trying to run on an old Galaxy S7 never asks for the dialog. I am using air 32, I might just roll back to an older version of air SDK to see if that helps,

**Reply by pinter75** [reply]
Ok, I am sure that the extension will still work, because if I flag that the permission is granted, the adapter state is null, but it invokes scanning. So if it is just the authorization not being set at runtime, is it worth my while just getting the permissions ane you guys do? That way I can still use my old ane, and the newer permissions ane to launch the bluetooth auth dialog?

**Reply by pinter75** [reply]
Managed to find a version of the android support ane, and it now works!

**Reply by marchbold** [reply]
Glad to hear. 

Is there any reason you aren't updating though? there have been a lot of changes since 2017 :)

**Reply by pinter75** [reply]
Because I have this one working :-)

Quick question, some of the Bluetooth HM10 adapters I am testing don't always report their name to Android. I wonder does the ane support mac address connections eg 00:15:83:00:71:82?
